rest-api-spec/src/main/resources/rest-api-spec/api/cat.thread_pool.json

```diff
@@ -44,7 +44,7 @@
           "p"
         ],
         "deprecated":{
```

Comment:
I hadn't originally back-ported the deprecation to to 7.7, but am on https://github.com/elastic/elasticsearch/pull/56737/files this will help reduce the conflict.